base_geoengine/static/src/js/views/geoengine/records_panel/records_panel.esm.js

```diff
@@ -0,0 +1,77 @@
+/** @odoo-module */
+
+/**
+ * Copyright 2023 ACSONE SA/NV
+ */
+import {useService} from "@web/core/utils/hooks";
+import {SearchBarRecords} from "./search_bar_records/search_bar_records.esm";
+
+const {Component, useState, onWillRender, onWillStart, onWillUpdateProps} = owl;
+
+export class RecordsPanel extends Component {
+    setup() {
+        super.setup();
+
+        this.state = useState({
+            isFolded: false,
+            isClicked: 0,
+            modelDescription: "",
+            records: [],
+        });
+        this.orm = useService("orm");
+        onWillStart(() => (this.state.records = this.props.list.records));
```

Comment:
this feels like a red flag to me. usually, duplicating props value in component state is wrong. I however don't know enough of your code to be sure yet

Reply:
maybe consider having `activeRecords`  getter in your class that simply return the props.records filtered by active filter, so you can remove `records` from your state (and probably need to add the filterValue to the state)